inc/dashboard/provider.class.php

```diff
@@ -1823,42 +1822,42 @@ private static function getFiltersCriteria(string $table = "", array $apply_filt
       }
 
       if (isset($apply_filters['user_tech'])) {
+         $users_id = null;
          if ((int) $apply_filters['user_tech'] > 0) {
             $users_id = (int) $apply_filters['user_tech'];
          } else if ($apply_filters['user_tech'] == 'myself') {
             $users_id = $_SESSION['glpiID'];
-         } else {
-            // Invalid value, should never happen
-            $users_id = -1;
          }
 
-         if ($DB->fieldExists($table, 'users_id_tech')) {
```

Comment:
Apply the filter only if $users_id !== null. all the code is indented 1 level deeper